File: backend/app/api/v1/endpoints/health.py
from fastapi import APIRouter

from app.core.config import get_settings
from app.schemas.health import HealthResponse, SystemInfoResponse

router = APIRouter()


@router.get("/health", response_model=HealthResponse)
def health() -> HealthResponse:
    return HealthResponse(status="ok", service="decilyra-api")


@router.get("/system", response_model=SystemInfoResponse)
def system_info() -> SystemInfoResponse:
    settings = get_settings()
    return SystemInfoResponse(
        service="decilyra-api",
        version="0.1.0",
        environment=settings.environment,
        database_configured=bool(settings.database_url),
        api_prefix=settings.api_v1_prefix,
    )

File: frontend/src/lib/api/health.ts
import { apiGet } from "@/lib/api/client";

export type HealthResponse = {
  status: string;
  service: string;
};

export type SystemInfoResponse = {
  service: string;
  version: string;
  environment: string;
  database_configured: boolean;
  api_prefix: string;
};

export function getHealth() {
  return apiGet<HealthResponse>("/api/v1/health");
}

export function getSystemInfo() {
  return apiGet<SystemInfoResponse>("/api/v1/system");
}
